src/drivers/mb12xx/mb12xx.cpp

```diff
@@ -453,7 +514,7 @@ MB12XX::read(struct file *filp, char *buffer, size_t buflen)
 		}
 
 		/* wait for it to complete */
-		usleep(MB12XX_CONVERSION_INTERVAL);
```

Comment:
Yes, split it in two functions and make the call to the 2nd one here, so that it will be called in cycling_rate*2 us. Using a usleep here blocks the work queue for everyone else.